syntax = "proto2";

package ZilliqaMessage;


message ProtoTxId
{
    required string txid = 1;
}

message ProtoTxHash
{
    required string txhash = 1;
}

message ProtoTxHashes
{
    required uint64 number = 1;
    repeated ProtoTxHash txhashes = 2;
}

message ProtoAddress
{
    required string address = 1;
}

message ProtoBlockNum
{
    required uint64 blocknum = 1;
}

message ProtoShardingStruct
{
    optional string error = 1;
    repeated uint32 numpeers = 2;
}

message ProtoBlockListing {
    optional string error = 1;
    optional int32 maxpages = 2;
    message BlockData {
        required string hash = 1;
        required int32 blocknum = 2;
    }
    repeated BlockData data = 3;
}

message ProtoPage
{
	required uint32 page = 1;
}

message ProtoBlockChainInfo
{
    required uint32 numpeers = 1;
    required string numtxblocks = 2;
    required string numdsblocks = 3;
    required string numtxns = 4;
    required double txrate = 5;
    required double txblockrate = 6;
    required double dsblockrate = 7;
    required uint64 currentminiepoch = 8;
    required uint64 currentdsepoch = 9;
    required uint32 numtxnstxepoch = 10;
    required string numtxnsdsepoch = 11;
    required ProtoShardingStruct shardingstructure = 12;
}
